fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java

```diff
@@ -65,12 +75,12 @@ default Table getTable(String dbName, String tblName) {
     }
 
     /**
-     * Return all partition names of the table.
-     * @param databaseName the name of the database
-     * @param tableName the name of the table
-     * @return a list of partition names
+     * Get the remote file information from hdfs or s3. It is mainly used to generate ScanRange for scheduling.
+     * @param table
+     * @param partitionKeys selected columns
+     * @return the remote file information of the query to scan.
      */
-    default List<String> getPartitionNames(String databaseName, String tableName) {
+    default List<RemoteFileInfo> getRemoteFileInfos(Table table, List<PartitionKey> partitionKeys) {
```

Comment:
why do we need this API

Reply:
this needed in building scanRange or HiveTable.tothrift

Reply:
This api isn't very common. I think we will refactor it later. Currently only hive and hudi are needed, I think it is enough at present.